### DISCUSSION TITLE
Subdivide pre existing grain list

### DISCUSSION THREAD

**Discussion by gautierromain** [mtex-toolbox/mtex#1186] [Ask Anything]
Hello everybody,
I am a PhD student and I am working with Mtex. However I have an issue that I can't find how to solve.

I have a pre existing grain list "grains_big". The Figure "1.jpg" shows the list "grain-big", which is colored regarding the orientation and the grains are separated by black lines. I want to subdivide these grains according to their orientation.

To do so I erase the EBSD data not contained into "grains_big" and then researched grains into the rest of EBSD data by using _calcGrains_. The new list of grains is named "grains_big2". However when I want to plot "grains_big2" I obtain the image "2.jpg". I wanted only the EBSD data within "grains_big" to be taken into account, but here I have the whole map...
I did the same operation but this time using the option _tight_ when using _calcGrains_ . This time I come up with the image "3.jpg" when I use _plot(grains_big2)_. It still doesn't take into account only the "grains_big" list.

I am sure there is a way but Mtex is quite new for me and I am not sure what to try next.
Thanks a lot for developing this powerful toolbox it's really great!!

Romain
![1](https://user-images.githubusercontent.com/89841030/131486124-8d0819e9-3551-4597-8bc2-36c72d075e36.jpg)
![2](https://user-images.githubusercontent.com/89841030/131486163-69a6f42c-c7f8-486f-a918-df40b9d5121c.jpg)
![3](https://user-images.githubusercontent.com/89841030/131486170-53ee7813-69b0-4269-b5ec-b03d3ed604b9.jpg)

**Comment by zmichels** [comment]
Hi Romain,

I think it would help if you could share the code you are using, so that others can perhaps see what needs fixing.

Cheers,
Zach

**Comment by gautierromain** [comment]
As @zmichels suggested, here's my code. I hope it is understable, otherwise let me know and I'll try to comment it more.
Thank you for your help.

Cheers

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  crystalSymmetry('m-3m', [4.096 4.096 4.096], 'mineral', 'Phase_1', 'color', [0.5,0.1,0.1])};

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','intoPlane');

%% Specify File Names

% path to files
pname = 'D:\Documents\These\Logiciel\Matlab\AFM_2\'
% which files to be imported
fname1 = [pname 'rotation_to_euler.ctf']
fname2 = [pname 'Al_Mg1_s3_3_030920_rot_cropped_denoised_trans2_cropped_denoised_rot_bc_bs.ctf']

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname1,CS,'interface','ctf',...
  'convertEuler2SpatialReferenceFrame');
rot = rotation.byAxisAngle(xvector,180*degree);
ebsd = rotate(ebsd,rot,'keepXY');

ebsd2 = EBSD.load(fname2,CS,'interface','ctf',...
  'convertEuler2SpatialReferenceFrame');
rot = rotation.byAxisAngle(xvector,180*degree);
ebsd2 = rotate(ebsd2,rot,'keepXY');

%% Refill grains
% reconstruct the grain structure
ebsd = ebsd.gridify;
[grains,ebsd('indexed').grainId] = calcGrains(ebsd('indexed'),'angle',1*degree);
ebsd(grains(grains.grainSize<=10)) = [];
[grains,ebsd('indexed').grainId] = calcGrains(ebsd('indexed'),'angle',2*degree);
F = splineFilter;
F = halfQuadraticFilter;
F.alpha = 0.25;
ebsd = smooth(ebsd('indexed'),F,'fill');
[grains,ebsd('indexed').grainId] = calcGrains(ebsd('indexed'),'angle',2*degree);



%% Cut the grain list in two parts
gr=grains(89).meanOrientation
grains_big = grains(angle(grains.meanOrientation,gr)./degree <10)
grains_small=grains(angle(grains.meanOrientation,gr)./degree >=10)

%% Grain Reconstruction ebsd2
[grains2,ebsd2.grainId] = calcGrains(ebsd2,'angle',10*degree);
grains2 = smooth(grains2,5);

%% Grain attribution 
%make the two same list of grains for the two ebsd maps
ebsd = ebsd.gridify;
ebsd2 = ebsd2.gridify;

grains2=grains

ebsd2.grainId=ebsd.grainId

%% IPF for mean grain orientation (Image "1.jpg")
figure(1)
ipfKey2 = ipfHSVKey(ebsd2('Phase_1').CS.Laue);
ipfKey2.inversePoleFigureDirection = vector3d.Z;
% this computes the colors for each pixel orientation 
colors2 = ipfKey2.orientation2color(ebsd2(grains_big).orientations);
plot(ebsd2(grains_big),colors2)
hold on
plot(grains_big.boundary,'linewidth',2)
hold off

%% separate
ebsd2(grains_small)=[]

%% Subgrains (Image "2.jpg")
figure(2)
[grains_big2,ebsd2('indexed').grainId] = calcGrains(ebsd2('indexed'),'angle',10*degree);
plot(grains_big2)
 
%% Subgrains2 (Image "3.jpg")
figure(3)
[grains_big2,ebsd2('indexed').grainId] = calcGrains(ebsd2('indexed'),'angle',10*degree,'boundary','tight');
 plot(grains_big2)
